dll/win32/browseui/addresseditbox.cpp

```diff
@@ -94,6 +95,73 @@ HRESULT STDMETHODCALLTYPE CAddressEditBox::SetCurrentDir(long paramC)
     return E_NOTIMPL;
 }
 
+BOOL CAddressEditBox::GetComboBoxText(CComHeapPtr<WCHAR>& pszText)
+{
+    pszText.Free();
+    INT cchMax = fCombobox.GetWindowTextLength() + sizeof(UNICODE_NULL);
+    if (!pszText.Allocate(cchMax))
+        return FALSE;
+    return fCombobox.GetWindowText(pszText, cchMax);
+}
+
+HRESULT CAddressEditBox::GetAbsolutePidl(PIDLIST_ABSOLUTE *pAbsolutePIDL)
+{
+    CComPtr<IBrowserService> isb;
+    HRESULT hr = IUnknown_QueryService(fSite, SID_STopLevelBrowser, IID_PPV_ARG(IBrowserService, &isb));
+    if (FAILED_UNEXPECTEDLY(hr))
+        return hr;
+
+    hr = isb->GetPidl(pAbsolutePIDL);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return hr;
+
+    return S_OK;
+}
+
+/* Execute command line from address bar */
+BOOL CAddressEditBox::ExecuteCommandLine()
+{
+    /* Get command line */
+    CComHeapPtr<WCHAR> pszCmdLine;
+    if (!GetComboBoxText(pszCmdLine))
+        return FALSE;
+
+    /* Split 1st parameter from trailing arguments */
+    PWCHAR args = PathGetArgsW(pszCmdLine);
+    if (args && *args)
+    {
+        --args;
+        *args = UNICODE_NULL;
+        ++args;
+    }
+
+    PathUnquoteSpacesW(pszCmdLine); /* Unquote the 1st parameter */
```

Comment:
This looks like unnecessary extra work. Shouldn't this code be just like the "Run" dialog and call the very same underlying shell function?

Reply:
`ShellExecCmdLine` is not exported.

Reply:
`_wsystem` is not implemented yet and it in Windows will create console window that is not applicable in this case.

Reply:
This looks ok for now,
in the future it should probably also check if it can navigate itself, or if it has to start a new instance.

Reply:
> `_wsystem` is not implemented yet [...]

`_system` or `_wsystem` must not be used in any case, so...